### PR TITLE
feat: #44 Feature current session profile

### PR DESCRIPTION
**Pull request by codedecoded01723**
1.Added multi device capabilities
2.Added current settings save option
3.Changes asked by tyler 
4.Added confirmation dialog
5.Added and implemented multi profile active 
6.Changed the  edit options for profile

### CODE REVIEW

**Review by tylerkron — CHANGES_REQUESTED** [review]
There are a handful of Codacy (static code analysis) issues. Please address those first, they should be pretty quick to fix. In the mean time I'll do some testing locally!

File: Daqifi.Desktop/View/AddprofileDialog.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            try
+            {
+                var data = SelectedDevice.SelectedItems;
+            }
+            catch (Exception ex)
+            {
+
+                AppLogger.Error(ex, "Error in updating ui of profile flyout");
+            }
+        }
```
Comment (tylerkron) [inline Daqifi.Desktop/View/AddprofileDialog.xaml.cs:46]: Is `SelectedDevice_PreviewMouseDown` used anywhere? It doesn't seem to do anything

File: Daqifi.Desktop/View/Flyouts/UpdateProfileFlyout.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            }
+
+        }
+        private void UpdatedProfileChannelListSelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (_isInitializing)
+            {
+                return;
+            }
+        }
```
Comment (tylerkron) [inline Daqifi.Desktop/View/Flyouts/UpdateProfileFlyout.xaml.cs:153]: Does `UpdatedProfileChannelListSelectionChanged` do anything?

File: Daqifi.Desktop/ViewModels/AddProfileDialogViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+
+        private ObservableCollection<IStreamingDevice> _availableDevices = new ObservableCollection<IStreamingDevice>();
+        public ObservableCollection<IStreamingDevice> AvailableDevices
+        {
+            get => _availableDevices;
+            set
+            {
+                _availableDevices = value;
+                //checkAddProfileButton();
```
Comment (tylerkron) [inline Daqifi.Desktop/ViewModels/AddProfileDialogViewModel.cs:41]: Can this commented code be deleted?

File: Daqifi.Desktop/ViewModels/AddProfileDialogViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        }
+
+        private ObservableCollection<IChannel> _availableChannels = new ObservableCollection<IChannel>();
+        public ObservableCollection<IChannel> AvailableChannels
+        {
+            get => _availableChannels;
+            set
+            {
+                _availableChannels = value;
+               // checkAddProfileButton();
```
Comment (tylerkron) [inline Daqifi.Desktop/ViewModels/AddProfileDialogViewModel.cs:53]: Can delete?

**Review by tylerkron — CHANGES_REQUESTED** [review]
Looking good! just a few comments

File: Daqifi.Desktop/ViewModels/AddProfileDialogViewModel.cs (hunk 3 above)
Comment (codedecoded01723) [reply Daqifi.Desktop/ViewModels/AddProfileDialogViewModel.cs:41]: done , fixed.

File: Daqifi.Desktop/View/AddprofileDialog.xaml.cs (hunk 1 above)
Comment (codedecoded01723) [reply Daqifi.Desktop/View/AddprofileDialog.xaml.cs:46]: done , fixed.

**Review by tylerkron — APPROVED** [review]

**Review by tylerkron — CHANGES_REQUESTED** [review]
I was doing some final testing. I deleted all my profiles (I deleted the profiles file too). After doing so, I wasn't able to create a new profile. 
image

**Review by tylerkron — CHANGES_REQUESTED** [review]
We are very close on this. Please prioritize getting this feature working before continuing with other features.

## Issue: Only 1 channel actually gets logged
When testing, there is a problem. I selected 5 analog channels. It looks like 5 are selected when I apply the profile. However, when I run the logging, only the first channel is actually logged.

Reproduction Steps
1. Create a new profile
2. Select channels AI0, AI1, AI2, AI3, AI4
3. Active the profile
4. Go the live graph
5. Turn logging on. Notice you only see one channel actually being logged
6. Stop logging
7. Find the session in Logged Data and verify that only one channel was logged
image


## Issue - Cant' set 1Hz as a frequency
When creating a profile, I can't set the frequency to 1Hz. 2Hz seems to be the minimum but 1Hz is valid
image
image


## Issue - Multiple Active Profiles
I should only be able to set one active at a time.
image

**Pull request closed without merging**